filter/partial_clone.go

```diff
@@ -0,0 +1,69 @@
+package filter
+
+import (
+	"fmt"
+	"sync/atomic"
+
+	"github.com/AdRoll/baker"
+)
+
+var PartialCloneDesc = baker.FilterDesc{
+	Name:   "PartialClone",
+	New:    NewPartialClone,
+	Config: &PartialCloneConfig{},
+	Help:   "Copy a list of fields to a new record and process this new record, discarding the original one",
+}
+
+type PartialCloneConfig struct {
+	Fields []string `help:"Fields that must be copied to the new line" required:"true"`
+}
+
+type PartialClone struct {
+	numProcessedLines int64
+	numFilteredLines  int64
+
+	fieldIdx     []baker.FieldIndex
+	createRecord func() baker.Record
+}
+
+func NewPartialClone(cfg baker.FilterParams) (baker.Filter, error) {
+	if cfg.DecodedConfig == nil {
+		cfg.DecodedConfig = &PartialCloneConfig{}
+	}
+	dcfg := cfg.DecodedConfig.(*PartialCloneConfig)
+
+	if len(dcfg.Fields) == 0 {
+		return nil, fmt.Errorf("PartialClone: add at least one field")
+	}
+
+	fieldIdx := []baker.FieldIndex{}
+	for _, f := range dcfg.Fields {
+		idx, ok := cfg.FieldByName(f)
+		if !ok {
+			return nil, fmt.Errorf("Can't resolve field name %s", f)
```

Comment:
nit: it's better for an error message to not start with caps in order to remain syntactically correct if it was to be wrapped with another before printed to the user